pkg/controller/job/job_controller.go

```diff
@@ -1239,7 +1243,11 @@ func (jm *Controller) enactJobFinished(job *batch.Job, finishedCond *batch.JobCo
 		metrics.JobFinishedNum.WithLabelValues(completionMode, "succeeded").Inc()
 	} else {
 		jm.recorder.Event(job, v1.EventTypeWarning, finishedCond.Reason, finishedCond.Message)
```

Comment:
We also don't want to record the event....

But why not just update the finishedCond earlier so that we do the update in this cycle already?

Reply:
it seems possible but when I tried to naively update the finishedCond it didn't work because the needsFlush was set to false after the first flush. then I thought that mutating the variable is just conceptually challenging, but indeed it seems doable.

Reply:
Updated, also don't use the FailureTarget for the path without finalizers.